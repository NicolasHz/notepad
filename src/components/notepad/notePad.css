.NotePad {
    height: 0;
    width: 0;
    z-index: 300;
    position: fixed;
    bottom: 7em;
    overflow-y: scroll;
    padding: 1em;
    border-radius: 5px;
    right: 10%;
    background: white;
    /* transition: all 0.4s cubic-bezier(.36, -0.64, .34, 1.76); */
}

.NotePad.enter {
    opacity: 0;
}

.NotePad.entered {
    opacity: 1;
    animation: grow .6s ease-out;
    /* width: 300px;
    height: 300px;
    transition: all 0.8s cubic-bezier(.36, -0.64, .34, 1.76); */
}

.NotePad.show {
    height: 50%;
    width: 70%;
    /* animation: grow .8s ease-in; */
}

.NotePad.hide {
    /* opacity: 0;
    margin: 0;
    height: 0; */
    animation: leave .8s ease-out;
    animation-direction: reverse;
    /* transition: all 0.8s cubic-bezier(.36, -0.64, .34, 1.76); */
}

.NotePad::-webkit-scrollbar {
    width: .8em;
}

.NotePad::-webkit-scrollbar-track {
    -webkit-box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.3);
}

.NotePad::-webkit-scrollbar-thumb {
    background-color: darkgrey;
    outline: 1px solid slategrey;
}

@keyframes grow {
    0% {
        opacity: 0;
        height: 5%;
        width: 10%;
    }

    20% {
        opacity: .2;
        height: 5%;
        width: 45%;
    }

    40% {
        opacity: .7;
        height: 5%;
        width: 55%;
    }

    60% {
        opacity: .9;
        height: 5%;
        width: 65%;
    }

    80% {
        opacity: 1;
        height: 5%;
        width: 70%;
    }

    100% {
        opacity: 1;
        height: 50%;
        width: 70%;
    }
}

@keyframes leave {
    0% {
        opacity: 0;
        height: 5%;
        width: 10%;
    }

    20% {
        opacity: .4;
        height: 5%;
        width: 70%;
    }

    40% {
        opacity: .6;
        height: 10%;
        width: 70%;
    }

    60% {
        opacity: .8;
        height: 20%;
        width: 70%;
    }

    80% {
        opacity: 1;
        height: 30%;
        width: 70%;
    }

    100% {
        opacity: 1;
        height: 50%;
        width: 70%;
    }
}


@media (min-width: 800px) {}