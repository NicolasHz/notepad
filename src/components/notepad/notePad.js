import React, { Component } from 'react';
import * as classes from './notePad.css'
// Redux
import { connect } from 'react-redux';
import * as actions from '../../store/actions/index';
import CSSTransition from "react-transition-group/CSSTransition";
// Components
import NoteList from '../notes-list/note-list'
class NotePad extends Component {

    constructor(props) {
        super();
        this.animationTiming = {
            enter: props.enterTime || 600,
            exit: props.exitTime || 800
        };
        this.state = { show: false }
    }

    componentDidMount() {
        setTimeout(() => {
            this.setState({ show: true })
        }, 5000);
    }

    render() {
        return (
            <CSSTransition
                mountOnEnter
                unmountOnExit
                in={this.state.show}
                timeout={this.animationTiming}
                classNames={{
                    enter: this.props.animation ? classes[this.props.animation.enter] : classes.enter,
                    enterActive: this.props.animation ? classes[this.props.animation.entered] : classes.entered,
                    enterDone: this.props.animation ? classes[this.props.animation.show] : classes.show,
                    exit: this.props.animation ? classes[this.props.animation.hide] : classes.hide
                }}>
                <div className={classes.NotePad}>
                    <NoteList deleteNote={(noteId) => this.props.onRemoveNote(noteId)} notes={this.props.notes} />
                </div>
            </CSSTransition>
        )
    }
}

const mapStateToProps = state => {
    return {
        notes: state.notesState.notes
    }
}

const mapDispatchToProps = dispatch => {
    return {
        onAddNote: (noteToAdd) => dispatch(actions.addNote(noteToAdd)),
        onRemoveNote: (noteId) => dispatch(actions.removeNote(noteId))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(NotePad);