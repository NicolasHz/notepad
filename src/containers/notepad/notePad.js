import React, { Component } from 'react';
import * as classes from './notePad.css'
// Redux
import { connect } from 'react-redux';
import * as actions from '../../store/actions/index';
import CSSTransition from "react-transition-group/CSSTransition";
// Components
import Auxiliar from '../../hoc/auxiliar/auxiliar';
import NotepadForm from './notepadForm/notepad-form';
import NoteList from '../../components/notes-list/note-list';
// Utility
import { delayedProps } from '../../shared/utility';
import Backdrop from '../../components/UI/backdrop/backdrop';

class NotePad extends Component {

    constructor(props) {
        super();
        this.animationTiming = {
            enter: props.enterTime || 600,
            exit: props.exitTime || 800
        };
        this.state = {
            show: false,
            notes: []
        }
        this.NotePadListRef = React.createRef();
    }

    componentWillReceiveProps(newProps) {
        delayedProps.call(this, newProps, 1000)
    }

    componentDidUpdate(prevState) {
        prevState.notes.length === this.state.notes.length &&
            setTimeout(() => {
                this.NotePadListRef.current && this.NotePadListRef.current.scrollTo({
                    top: this.NotePadListRef.current.scrollHeight,
                    behavior: "smooth"
                });
            }, 800);
    }

    render() {
        console.log(this.props)
        return (
            <Auxiliar>
                <CSSTransition
                    mountOnEnter
                    unmountOnExit
                    in={this.props.showNotepad}
                    timeout={this.animationTiming}
                    classNames={{
                        enter: this.props.animation ? classes[this.props.animation.enter] : classes.enter,
                        enterActive: this.props.animation ? classes[this.props.animation.entered] : classes.entered,
                        enterDone: this.props.animation ? classes[this.props.animation.show] : classes.show,
                        exit: this.props.animation ? classes[this.props.animation.hide] : classes.hide
                    }}>
                    <div className={classes.NotePad} style={{ top: this.props.top || "unset", right: this.props.right || "5vmax;", bottom: this.props.bottom || "7em", left: this.props.left || "unset" }}>
                        <div className={classes.NotePadListWrapper} ref={this.NotePadListRef}>
                            <NoteList showNoteErrors deleteNote={(noteId) => this.props.onRemoveNote(noteId)} notes={this.state.notes} />
                        </div>
                        <div className={classes.NotePadFormWrapper}>
                            <NotepadForm onAddNoteHandler={(note) => { this.props.onAddNote(note) }} />
                        </div>
                    </div>
                </CSSTransition>
                {this.props.showNotepad && <Backdrop clicked={() => { this.props.onHideNotepad() }} />}
            </Auxiliar>
        )
    }
}

const mapStateToProps = state => {
    return {
        notes: state.notesState.notes,
        showNotepad: state.UIState.show
    }
}

const mapDispatchToProps = dispatch => {
    return {
        onAddNote: (noteToAdd) => dispatch(actions.addNote(noteToAdd)),
        onRemoveNote: (noteId) => dispatch(actions.removeNote(noteId)),
        onHideNotepad: () => dispatch(actions.hideNotePad())
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(NotePad);