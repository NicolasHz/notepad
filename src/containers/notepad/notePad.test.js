import React from 'react';
import ReactDOM from 'react-dom';
import { NotePad } from './notePad';
import renderer from 'react-test-renderer';

const notesMock = [{
    message: "Remember buy carrots?",
    id: 1,
    errors: []
}];
describe('Notepad works', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <NotePad />
            , div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('should change with a new note', () => {
        let component = renderer.create(<NotePad showNotepad={true} notes={[]}/>)
        let jsonTree = component.toJSON();
        expect(jsonTree).toMatchSnapshot();
        
        component = renderer.create(<NotePad showNotepad={true} notes={notesMock}/>)
        jsonTree = component.toJSON();
        expect(jsonTree).toMatchSnapshot();
    });
});