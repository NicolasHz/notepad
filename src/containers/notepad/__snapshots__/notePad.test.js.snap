// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notepad works should change with a new note 1`] = `
Array [
  <div
    style={
      Object {
        "bottom": "7em",
        "left": "unset",
        "right": "5vmax",
        "top": "unset",
      }
    }
  >
    <div>
      <div
        className=" "
      />
    </div>
    <div>
      <form
        onSubmit={[Function]}
      >
        <div>
          <label
            className=""
          >
            
          </label>
          <input
            className=""
            onChange={[Function]}
            placeholder="Your note here"
            type="text"
            value=""
          />
        </div>
        <button
          disabled={false}
        >
          Add Note
        </button>
      </form>
    </div>
  </div>,
  <div
    onClick={[Function]}
  />,
]
`;

exports[`Notepad works should change with a new note 2`] = `
Array [
  <div
    style={
      Object {
        "bottom": "7em",
        "left": "unset",
        "right": "5vmax",
        "top": "unset",
      }
    }
  >
    <div>
      <div
        className=" "
      >
        <div>
          <p>
            Remember buy carrots?
          </p>
          <span
            onClick={[Function]}
          />
        </div>
        <div />
      </div>
    </div>
    <div>
      <form
        onSubmit={[Function]}
      >
        <div>
          <label
            className=""
          >
            
          </label>
          <input
            className=""
            onChange={[Function]}
            placeholder="Your note here"
            type="text"
            value=""
          />
        </div>
        <button
          disabled={false}
        >
          Add Note
        </button>
      </form>
    </div>
  </div>,
  <div
    onClick={[Function]}
  />,
]
`;
